src/prefect/task_worker.py

```diff
@@ -174,8 +187,20 @@ async def start(self) -> None:
                 else:
                     raise
 
-    @sync_compatible
-    async def stop(self):
+    async def astop(self):
+        """Stops the task worker's polling cycle."""
+        if not self.started:
+            raise RuntimeError(
+                "Task worker has not yet started. Please start the task worker by"
+                " calling .start()"
+            )
+
+        self._started_at = None
+        self.stopping = True
+
+        raise StopTaskWorker
+
+    def stop(self):
```

Comment:
Do we want an `@async_dispatch` here?

Reply:
my fault, I went to untangle the sync_compatible stuff and ran into trouble because `TaskWorker` needs to be instantiated in an async context, but didn't fully clean up after myself apparently, will update